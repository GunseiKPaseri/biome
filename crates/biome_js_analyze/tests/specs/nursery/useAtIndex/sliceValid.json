[
  "array.slice(-1)",
  "new array.slice(-1)",
  "array.slice(-1)?.[0]",
  "array.slice?.(-1)[0]",
  "array?.slice(-1)[0]",
  "array.notSlice(-1)[0]",
  "array.slice()[0]",
  "array.slice(...[-1])[0]",
  "array.slice(-1).shift?.()",
  "array.slice(-1)?.shift()",
  "array.slice(-1).shift(...[])",
  "new array.slice(-1).shift()",
  "array.slice(-1)[0] += 1",
  "++ array.slice(-1)[0]",
  "array.slice(-1)[0] --",
  "delete array.slice(-1)[0]",
  "array.slice(-9.1, -8.1)[0]",
  "array.slice(-unknown, -unknown2)[0]",
  "array.slice(-9.1, unknown)[0]",
  "array.slice(-9, unknown).pop()",
  "array.slice(-9, ...unknown)[0]",
  "array.slice(...[-9], unknown)[0]",
  "array.slice(-9, unknown)[0]",
  "array.slice(-9, unknown).shift()",
  "const KNOWN = -8; array.slice(-9, KNOWN).shift()",
  "(( (( array.slice( ((-9)), ((unknown)), ).shift ))() ));",
  "array.slice(-9, (a, really, _really, complicated, second) => argument)[0]"
]
