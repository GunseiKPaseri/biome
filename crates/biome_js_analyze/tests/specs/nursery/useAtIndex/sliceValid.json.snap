---
source: crates/biome_js_analyze/tests/spec_tests.rs
assertion_line: 86
expression: sliceValid.json
---
# Input
```cjs
array.slice(-1)
```

# Input
```cjs
new array.slice(-1)
```

# Input
```cjs
array.slice(-9).pop()
```

# Input
```cjs
array.slice(-1.1)[0]
```

# Input
```cjs
array.slice(-1)?.[0]
```

# Input
```cjs
array.slice?.(-1)[0]
```

# Input
```cjs
array?.slice(-1)[0]
```

# Input
```cjs
array.notSlice(-1)[0]
```

# Input
```cjs
array.slice()[0]
```

# Input
```cjs
array.slice(...[-1])[0]
```

# Input
```cjs
array.slice(-1).shift?.()
```

# Input
```cjs
array.slice(-1)?.shift()
```

# Input
```cjs
array.slice(-1).shift(...[])
```

# Input
```cjs
new array.slice(-1).shift()
```

# Input
```cjs
array.slice(-1)[0] += 1
```

# Input
```cjs
++ array.slice(-1)[0]
```

# Input
```cjs
array.slice(-1)[0] --
```

# Input
```cjs
delete array.slice(-1)[0]
```

# Input
```cjs
array.slice(-9.1, -8.1)[0]
```

# Input
```cjs
array.slice(-unknown, -unknown2)[0]
```

# Input
```cjs
array.slice(-9.1, unknown)[0]
```

# Input
```cjs
array.slice(-9, unknown).pop()
```

# Input
```cjs
array.slice(-9, ...unknown)[0]
```

# Input
```cjs
array.slice(...[-9], unknown)[0]
```

# Input
```cjs
array.slice(-9, unknown)[0]
```

# Input
```cjs
array.slice(-9, unknown).shift()
```

# Input
```cjs
const KNOWN = -8; array.slice(-9, KNOWN).shift()
```

# Input
```cjs
(( (( array.slice( ((-9)), ((unknown)), ).shift ))() ));
```

# Input
```cjs
array.slice(-9, (a, really, _really, complicated, second) => argument)[0]
```
